src/wazuh_db/wdb_global.c

```diff
@@ -1415,6 +1415,271 @@ cJSON* wdb_global_get_agents_by_connection_status (wdb_t *wdb, int last_agent_id
     return result;
 }
 
+int wdb_global_create_backup(wdb_t* wdb, char* output) {
+    char path[PATH_MAX-3] = {0};
+    char path_compressed[PATH_MAX] = {0};
+    char* timestamp = NULL;
+    DIR* dp = NULL;
+    int result = OS_INVALID;
+
+    dp = opendir(WDB_BACKUP_FOLDER);
+
+    if(!dp) {
+        mdebug1("Unable to open backup directory '%s' for Wazuh-DB", WDB_BACKUP_FOLDER);
+        snprintf(output, OS_MAXSTR + 1, "err Unable to open backup directory '%s' for Wazuh-DB", WDB_BACKUP_FOLDER);
+        return OS_INVALID;
+    }
+    closedir(dp);
```

Comment:
Just wondering. Which is the reason for opening the directory and closing it immediately after checking that it was successfully opened?

Reply:
You are right, it isn't necessary.
It was removed from `wdb_global_restore_backup()` and `wdb_global_create_backup()`.